source/mainnet/smart-contracts/tutorials/piggy-bank/deploying.rst

```diff
@@ -0,0 +1,160 @@
+
+.. code-block:: console
+
+   $./concordium-client contract init piggy_bank_part2_module --sender <account-name> --contract PiggyBank --name piggy_bank_part2_instance --energy 1000 --grpc-port 10001
+
+.. Note::
+   1000 is enough energy to initialize the piggy bank smart contract but you might need more energy for larger smart contracts.
+
+
+If everything works correctly the output has a green line with your smart contract instance index.
+
+.. code-block:: console
+
+   Contract successfully initialized with address: {"index":<smart-contract-instance-index>,"subindex":0}
+   Contract address {"index":<smart-contract-instance-index>,"subindex":0} was successfully named 'piggy_bank_part2_instance'.
+
+
+You can find additional information about initializing a smart contract instance in this :ref:`guide<initialize-contract>`.
+
+
+Updating the piggy bank smart contract (sending CCD)
+====================================================
+
+Let's deposit 1 CCD into the piggy bank to test if you can update the smart contract instance.
+
+.. code-block:: console
+
+   $./concordium-client contract update piggy_bank_part2_instance --entrypoint insert --energy 1000 --sender <account-name> --amount 1 --grpc-port 10001
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Successfully updated contract instance {"index":<smart-contract-instance-index>,"subindex":0} ('piggy_bank_part2_instance') using the function 'insert'.
+
+You can find additional information about updating a smart contract instance in this :ref:`guide<interact-instance>`.
+
+
+Viewing the piggy bank state
+============================
+
+You can display the raw return value of your piggy bank smart contract with the following command.
+
+.. code-block:: console
+
+   $./concordium-client contract invoke piggy_bank_part2_instance --entrypoint view --energy 1000 --grpc-port 10001
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Invocation resulted in success:
+   - Energy used: 510 NRG
+   - Return value (raw):
+   [0,64,66,15,0,0,0,0,0]
+   .
+
+
+You can find additional information about invoking in this :ref:`guide<invoke-instance>`.
+
+
+Updating the piggy bank smart contract (sending no CCD)
+=======================================================
+
+Let's smash the piggy bank contract to test that you can update the smart contract instance without sending CCD.
+
+.. code-block:: console
+
+   $./concordium-client contract update piggy_bank_part2_instance --entrypoint smash --sender <account-name> --energy 2000 --grpc-port 10001
+
+.. Note::
+   Ensure that you don't send any CCD to the smart contract instance and that the <account-name> is the same as the <account-name> that initialized the piggy bank smart contract instance.
```

Comment:
If you keep this section, then you should mention that this is because the smash entrypoint isn't payable and link to the section where we describe that attribute: https://developer.concordium.software/en/mainnet/smart-contracts/tutorials/piggy-bank/writing.html#inserting-ccd